lib/ja_serializer/dsl.ex

```diff
@@ -0,0 +1,424 @@
+defmodule JaSerializer.DSL do
+  @moduledoc """
+  A DSL for defining JSON-API.org spec complaint payloads.
+
+  Built on top of the `JaSerializer.Serializer` behaviour.
+
+  The following macros are available:
+
+    * `location/1` - Define the url of a single serialized object.
+    * `attributes/1` - Define the attributes to be returned.
+    * `has_many/2` - Define a has_many relationship.
+    * `has_one/2` - Define a has_one or belongs_to relationship.
+
+  This module should always be used in conjunction with
+  `JaSerializer.Serializer`, see `JaSerializer` for the best way to do so.
+
+  When `use`ing this module default implementations of the `links/2`,
+  `attributes/2`, and `relationships/2` will be defined on your module.
+
+  Overriding these callbacks can be a great way to customize your serializer
+  beyond what the DSL provides.
+
+  ## Example
```

Comment:
The previous paragraph talks about overriding callbacks, but this example does not do that. I was expecting the example to demonstrate that. Would it make sense to include that in the example?